js/components/discoveryPanel.js

```diff
@@ -121,12 +146,12 @@ var DiscoveryPanel = React.createClass({
 
     // Build discovery content blocks
     var discoveryContentBlocks = [];
-    for (var i = 0; i < relatedVideos.length; i++) {
+    for (var i = 0; i < relatedVideoPage.length; i++) {
```

Comment:
relatedVideos is the original array of all videos.

relatedVideoPage is the sliced paginated array of videos to be displayed per page.